### PR TITLE
bazaar menu changes everyday, cosmetics are dyeable and weareable by slimeoids

### PR DESCRIPTION
**Pull request by MeatyLearner**
- The bazaar's menu changes at 6am everyday
- Cometics are now dyeable and can be adorned by slimeoids (max is height)

### CODE REVIEW

File: ewmarket.py [hunk 1]
```diff
@@ -65,6 +69,22 @@ def __init__(self, id_server = None):
 					self.decayed_slimes = result[6]
 					self.donated_slimes = result[7]
 					self.donated_poudrins = result[8]
+
+					cursor.execute("SELECT {}, {} FROM bazaar_wares WHERE {} = %s".format(
```
Comment (magicksRampage) [inline ewmarket.py:73]: please update the README with the new database table

File: ewcmd.py [hunk 2]
```diff
@@ -251,9 +252,9 @@ def gen_data_text(
 			response += " {}".format(trauma.str_trauma_self)
 		
 		response_block = ""
-		for mutation in mutations:
-			mutation_flavor = ewcfg.mutations_map[mutation]
-			response_block += "{} ".format(mutation_flavor.str_describe_self)
+		#for mutation in mutations:
+		#	mutation_flavor = ewcfg.mutations_map[mutation]
+		#	response_block += "{} ".format(mutation_flavor.str_describe_self)
```
Comment (magicksRampage) [inline ewcmd.py]: put this back in

File: ewcmd.py [hunk 3]
```diff
@@ -224,7 +224,8 @@ def gen_data_text(
 		for cosmetic in cosmetics:
 			cos = EwItem(id_item = cosmetic.get('id_item'))
 			if cos.item_props['adorned'] == 'true':
-				adorned_cosmetics.append(cosmetic.get('name'))
+				hue = ewcfg.hue_map.get(cos.item_props.get('hue'))
+				adorned_cosmetics.append((hue.str_name + " colored " if hue != None else "") + cosmetic.get('name'))
```
Comment (magicksRampage) [inline ewcmd.py:229]: we should make this change for when you !data other people, too. see gen_data_text()

File: ewcosmeticitem.py [hunk 4]
```diff
@@ -107,3 +107,58 @@ def __init__(
...
+				cosmetic_item = EwItem(id_item=cosmetic.get("id_item"))
+				dye_item = EwItem(id_item=dye.get("id_item"))
+
+				hue = ewcfg.hue_map.get(dye_item.item_props.get('id_item'))
+
+				response = "You dye your {} in {} paint!".format(cosmetic_item.item_props.get('cosmetic_name'), hue.str_name)
+				cosmetic_item.item_props['hue'] = hue.id_hue
+
+				cosmetic_item.persist()
+				#ewitem.item_delete(id_item=dye.id_item)
```
Comment (magicksRampage) [inline ewcosmeticitem.py]: we should delete the dye. (it's dye.get("id_item"))

File: ewmarket.py [hunk 5]
```diff
@@ -107,6 +127,25 @@ def persist(self):
 				self.donated_poudrins,
 			))
 
+			# Remove all existing bazaar items
+			cursor.execute("DELETE FROM bazaar_wares WHERE {} = %s".format(
+				ewcfg.col_id_server
+			), (
+				self.id_server,
+			))
```
Comment (magicksRampage) [inline ewmarket.py:135]: lol, just use REPLACE INTO

File: ewslimeoid.py [hunk 6]
```diff
@@ -2761,3 +2778,60 @@ def __init__(
...
 
+async def dress_slimeoid(cmd):
+	user_data = EwUser(member = cmd.message.author)
+	slimeoid = EwSlimeoid(member = cmd.message.author)
+
+	if user_data.life_state == ewcfg.life_state_corpse:
+		response = "Slimeoids don't fuck with ghosts."
+
+	elif slimeoid.life_state == ewcfg.slimeoid_state_none:
+		response = "Wow, great idea! Too bad you don’t even have a slimeoid with which to saturate! You’d think you’d remember really obvious stuff like that, but no. What a dumbass."
```
Comment (magicksRampage) [inline ewslimeoid.py]: update flavor text

File: ewslimeoid.py [hunk 7]
```diff
@@ -2761,3 +2778,60 @@ def __init__(
...
+				# Remove hat
+				if cosmetic.item_props.get('slimeoid') == 'true':
+					response += "take the {} back from {}".format(cosmetic.item_props.get('cosmetic_name'), slimeoid.name)
+					cosmetic.item_props['slimeoid'] = 'false'
+				# Give hat
+				else:
+					# Remove hat from player if adorned
+					if cosmetic.item_props.get('adorned') == 'true':
+						cosmetic.item_props['adorned'] = 'false'
+						response += "take off your {} and give it to {}.".format(cosmetic.item_props.get('cosmetic_name'), slimeoid.name)
```
Comment (magicksRampage) [inline ewslimeoid.py:2850]: good thinking. it should work the same way when you try to adorn a cosmetic your slimoid is wearing. slimeoid cosmetics should also be dedorned when your slimeoid dies in combat or you dissolve it.

**Review by magicksRampage — CHANGES_REQUESTED** [review]

File: ewcmd.py (hunk 3 above)
Comment (MeatyLearner) [reply ewcmd.py:229]: done

File: ewslimeoid.py (hunk 7 above)
Comment (MeatyLearner) [reply ewslimeoid.py:2850]: done

**Pull request merged**